File: python-backend/src/functions/get_todo.py
from src.utils.package import package
from src.utils.dynamodb_utils import todosTableName, ddb_client

def handler(event, context):
    id = event['pathParameters']['id']

    todo_ddb = ddb_client.get_item(
        TableName=todosTableName,
        Key={'todosId': {'S': id}}
    )

    return package(200, todo_ddb['Item'])

File: node-backend/src/functions/getTodo.js
import { formatJSONResponse } from '../utils/utils.js'
import { ddbClient, todosTableName } from '../utils/dynamodbUtils.js'

export const handler = async (event) => {
  const id = event.pathParameters.id

  const todoDdb = await ddbClient.get({
    TableName: todosTableName,
    Key: { todosId: id }
  }).promise()

  if (!todoDdb.Item)
    return formatJSONResponse(404, { message: 'Id does not exist' })

  return formatJSONResponse(200, { todoObj: todoDdb.Item, id })
}
